### ISSUE TITLE
[1.8.1][windows_guest] puppet.working_directory fails on Windows

### ISSUE THREAD

**Issue by lmayorga1980** [mitchellh/vagrant#6762]
I am using Puppet 4 with `environment.conf` and everything looks good in Linux but it seems that is not working properly on Windows.

`VAGRANT_LOG=debug vagrant provision` does not show the specific `puppet apply` on Windows
`export WINRM_LOG=debug` printed most of the SOAP messages but it seems that I can only decode de Base64 encoded output from `:stdout` and `:stderr`.

Had to edit this file `./opt/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/communicators/winrm/communicator.rb`  to show the EncodedCommand.

It seems like `puppet.working_directory = 'C:/tmp/vagrant-puppet/environments/development'` or  `puppet.working_directory = '/tmp/vagrant-puppet/environments/development'` is breaking powershell.

```log
cd C:/tmp/vagrant-puppet/environments/development; if (`$?) {  puppet apply --verbose --detailed-exitcodes --environmentpath /tmp/vagrant-puppet/environments/ --environment development /tmp/vagrant-puppet/environments/development/manifests }
##########################
==> ruby_win2012r2: #< CLIXML
==> ruby_win2012r2: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><S S="Error">$? : The term '$?' is not recognized as the name of a cmdlet, function, script _x000D__x000A_</S><S S="Error">file, or operable program. Check the spelling of the name, or if a path was _x000D__x000A_</S><S S="Error">included, verify that the path is correct and try again._x000D__x000A_</S><S S="Error">At line:1 char:96_x000D__x000A_</S><S S="Error">+ ... velopment; if (`$?) {  puppet apply --verbose --detailed-exitcodes _x000D__x000A_</S><S S="Error">--environment ..._x000D__x000A_</S><S S="Error">+                    ~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : ObjectNotFound: ($?:String) [], CommandNotFoundE _x000D__x000A_</S><S S="Error">   xception_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : CommandNotFoundException_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
```

Without the `puppet.working_directory` it seems to work fine...

```log
puppet apply --verbose --detailed-exitcodes --environmentpath /tmp/vagrant-puppet/environments/ --environment development /tmp/vagrant-puppet/environments/development/manifests
```

Maybe @sneal  can help...

**Comment by sneal** [comment]
Does the cd command by itself work from a powershell terminal? I assume it does.

I'm not sure why there's an escape backtick before $?. Maybe try it without.

**Comment by lmayorga1980** [comment]
The issue happens only when I set the working_directory attribute either with  `/tmp/` or `C:\tmp` syntax.

**Comment by sneal** [comment]
Fixed by #6816
